### PR TITLE
Add internal solution for GCD and MST (cf-1513d) 

### PR DESCRIPTION
**Pull request by ChuangSheep**
~~expl probably comes before next Sat.~~ Done

_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: solutions/gold/cf-1513D.mdx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	// the edges that still have to be added have the cost p
+	ans += (ll)edges_remaining * p;
+
+	cout << ans << endl;
+}
+
+int main() {
+	int t;
+	cin >> t;
+	while (t-- > 0) { solve(); }
```
Comment (SansPapyrus683) [inline solutions/gold/cf-1513D.mdx:100]: for loop & put `solve()` in here

File: solutions/gold/cf-1513D.mdx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+Like [Kruskal's Algorithm](https://usaco.guide/gold/mst#kruskals), we generate
+our MST by adding edges greedily. Except the edges between any two adjacent
+elements in the array which cost $p$, an edge only exists between $i$ and $j$ if
+the greatest common divisor of all elements in $[i, j]$ equals the minimum of
+the range $[i, j]$.
+
+As the cost of these edges is the minimum element of the range, let's iterate
+through all elements of the array in ascending order. This way, we guarantee
+that we always add edges that cost as less as possible. Let the current element
```
Comment (SansPapyrus683) [inline solutions/gold/cf-1513D.mdx:20]: > add edges that cost as less as possible

*add edges of minimum cost

File: solutions/gold/cf-1513D.mdx (hunk 1 above)
Comment (SansPapyrus683) [inline solutions/gold/cf-1513D.mdx:100]:
```suggestion
	int test_num;
	cin >> test_num;
	for (int t = 0; t < test_num; t++) { solve(); }
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by SansPapyrus683 — APPROVED** [review]

**Review by ryanchou-dev — APPROVED** [review]
yessir

File: solutions/gold/cf-1513D.mdx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+	     [&](const int i, const int j) { return arr[i] < arr[j]; });
+
+	ll ans = 0;
+	int edges_remaining = n - 1;
+	// is_connected[i] == true if there is an edge between i and i+1
+	vector<bool> is_connected(n, false);
+	for (int i = 0; i < n; i++) {
+		int j = indices[i];
+		int val = arr[j];
+		/**
```
Comment (SansPapyrus683) [inline solutions/gold/cf-1513D.mdx:69]:
```suggestion
		/*
```

**Pull request merged**